### PR TITLE
Use organization Prettier config

### PR DESCRIPTION
**Pull request by davwheat**
**Changes proposed in this pull request:**

Moves our Prettier config from being handled locally in repositories to a central source of truth.

**Reviewers should focus on:**

Ensure formatting CI workflow runs successfully!

**Required changes:**

- [ ] Related core extension PRs: (need to be created)

### CODE REVIEW

**Comment by davwheat** [comment]
Performed a manual workflow run with Prettier here: https://github.com/flarum/core/actions/runs/1043251829

**Comment by davwheat** [comment]
After 3 too many commits, issue is fixed: https://github.com/flarum/core/runs/3099093166?check_suite_focus=true

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -23,6 +23,10 @@ jobs:
         with:
           node-version: "14"
 
+      - name: Install JS dependencies
+        run: npm i
+        working-directory: ./js
+
       - name: Check JS formatting
-        run: npx prettier --check src
+        run: ./node_modules/.bin/prettier --check src
```
Comment (datitisev) [inline .github/workflows/lint.yml:31]: This should use the package.json command, `npm run format-check`

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -23,6 +23,10 @@ jobs:
         with:
           node-version: "14"
 
+      - name: Install JS dependencies
+        run: npm i
```
Comment (datitisev) [inline .github/workflows/lint.yml:27]: This can be replaced with `npm ci` I think?

Comment (davwheat) [reply .github/workflows/lint.yml]: Indeed it can! :P

File: .github/workflows/lint.yml (hunk 1 above)
Comment (davwheat) [reply .github/workflows/lint.yml]: Ahh, I forget we have that. Updated

**Review by datitisev — APPROVED** [review]

**Comment by davwheat** [comment]
Rebased to fix PHP tests

**Review by SychO9 — APPROVED** [review]

**Pull request merged**